genomeGraph/genomeGraph.go

```diff
@@ -7,26 +7,27 @@ import (
 	"github.com/vertgenlab/gonomics/dnaTwoBit"
 	"github.com/vertgenlab/gonomics/exception"
 	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/numbers"
 	"io"
+	"log"
 	"strings"
 )
 
 // GenomeGraph struct contains a slice of Nodes
 type GenomeGraph struct {
-	Nodes []*Node
+	Nodes []Node
 }
 
 // Node is uniquely definded by Id and is encoded with information
 // describing sequence order and orientation and annotated variance
 type Node struct {
-	Id uint32
-	//Name      string
+	Id        uint32
+	ColumnId  uint32
```

Comment:
Right now Node Id will be given at input and never change.  There is also a rule that a nodeId can not point to a nodeId of a lesser value (error if the file does this).
ColumnId will be used such that two nodes having the same columnId are "aligned."  This may be a C and T node if there is a C and T SNP.  Or it could be C and T nodes that are aligned across graphs (e.g. human graph has a C and chimp graph has a T)

Reply:
Ooo yes yes yes. Thank you for the reminder!
I have a question about Node Id then. Considering inversions within a population, isn't it possible for a Node Id to point to a lower number? Or for these instances would we likely just have new nodes to represent this type of case? And I guess too I think about other scenarios where the Node Id order could be non-increasing, say when we are considering graphs between species. Could translocations break this rule maybe? Trying to convince myself that this rule of no nodeId to a lesser value will hold up biologically too.

Reply:
O wow, this is awesome! I dont think I've talked with anyone about these three pieces that you just highlighted! And makes sense even more now because I realize that inversions when drawn tend to be oversimplified. Like repeats, inversions can have internal polymorphisms from the "original" sequence. For example, the sequence is inverted in addition to containing  internals SNPs, indels, etc. within the inversion boundaries. So carving a new path and keeping things linear makes a lot of sense. I guess too there are other techniques out there for if we wanted to resolve the inversions between individuals or species if it isnt obvious via our graphs. And about to read a bit on DAG algorithms.